### DISCUSSION TITLE
IC2/C2I Adapters

### DISCUSSION THREAD

**Discussion by jhaprabhatt** [async-profiler/async-profiler#753] [Q&A] [answered]
I am profiling an application with event as wall. In the generated flame graph, I see for one call:

1. IC2/C2I Adapters 
2. g1_post_barrier_slow

Code flow is like below:

MyClasss

      Method 1 ---- At this point, report shows IC2/2CI adapter

          On some condition:

           Call Method 2 with 2 parameters, which are then stored in the instance variables--- At this point, report shows g1_post_barrier_slow

Google results haven't helped much, any pointers please ?

**Comment by apangin** [comment]
Can you please clarify what the quesion is exactly? What problem are you trying to solve?
If you can show the problematic flame graph (or at least a part of it), this would be helpful. Thanks.

**Reply by jhaprabhatt** [reply, marked as answer]
I was measuring wall time. I found the issue. One of my classes was not getting compiled due to size and once I got rid of it. These two issues went away.
